### PR TITLE
Adds SEP for decoupling Gen and Run

### PR DESCRIPTION
**Pull request by chuckha**
Signed-off-by: Chuck Ha <chuck@heptio.com>

Adds a sonobuoy enhancement proposal for improved customization of generated manifest.

### CODE REVIEW

File: docs/enhancements/decouple-gen-and-run.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+### User Stories
+
+#### Sonobuoy Scanner developer
+
+As a third party developer I have some post-processing tasks I'd like to
+accomplish in the aggregator server's pod. For example, it would be cool if I
+could add the [namespace deleter][1] pod to run after the aggregation server has
+run. I might also add in another container to ship the results tarball
+somewhere.
```
Comment (liztio) [inline docs/enhancements/decouple-gen-and-run.md]: We need a user story for`sonobuoy` cli users too

File: docs/enhancements/decouple-gen-and-run.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+### Goals
+
+* Allow clients of Sonobuoy to customize the generated manifest and then run it.
+
+### Non Goals
+
+* Make the customizations built in to sonobuoy via config or command line flags.
+
+## Proposal
```
Comment (liztio) [inline docs/enhancements/decouple-gen-and-run.md]: Do you have a preferred alternative?

File: docs/enhancements/decouple-gen-and-run.md (hunk 1 above)
Comment (chuckha) [reply docs/enhancements/decouple-gen-and-run.md]: Supporting object customization via the CLI is out of scope for this SEP. Upon thinking how to make this user story more generic, I could rename "Sonobuoy Scanner Developer" to "Sonobuoy client interface consumer"

File: docs/enhancements/decouple-gen-and-run.md (hunk 2 above)
Comment (chuckha) [reply docs/enhancements/decouple-gen-and-run.md]: Added https://github.com/heptio/sonobuoy/pull/413/files#diff-5644e2d9e24e88115ee8ba37fdca29e6R21

**Pull request closed without merging**